README_developers.md

```diff
@@ -91,128 +85,356 @@ $ # For a fix
 $ git checkout master
 $ git checkout -b my-bug-fix
 ```
+- Provide extended description in pull request
+- Reference the issue in the description
+- Keep pull requests focused on single issue
+- Whenever possible squash all commits on merge
+
+
+
+# Style Guidelines
+## Indentation
+
+### Indent with spaces
+Use two spaces for indentation, absolutely no tab characters.
+
+Rationale: Different editors may interpret tabs differently. Using spaces
+only is more likely to provide consistent look in different editors.
+
+### Macros
+Do not indent macros, even when they are nested.
+```c++
+// Yes:
+#ifdef MACRO1
+  a = 0;
+#ifdef MACRO2
+  b = 1;
+#endif // MACRO2
+#else  // MACRO1
+  a = 1;
+#endif //MACRO1
+
+// No:
+#ifdef MACRO1
+  a = 0;
+  #ifdef MACRO2
+    b = 1;
+  #endif // MACRO2
+#else  // MACRO1
+  a = 1;
+#endif //MACRO1
+```
+### Indentation of access specifiers
 
-## Indentation, Braces, and Declarations
+Prefer no indentation for `private`, `public`, and `protected` access
+specifiers.
+```c++
+// Yes: public and private keywords aligned with open/close braces 
+class Myclass
+{
+public:
+  int my_method();
 
-- Use two spaces for indentation, absolutely no tab characters.
-- No spaces between `if` and `(`
-- Avoid condition and loop bodies in the same line of code
-- Allman style braces
-- RAJA Lambdas
-  - Indent two spaces above surrounding scope
-  - No spaces around template parameter
-  - Line break before range segment
-  - Use `RAJA_LAMBDA` in favor of using `__device__` directly
-  - Use `hiop_*_policy` over directly using cuda or omp policies
-  - Prefer `RAJA::Index_type` over `int` or anything else
-- Prefer braces for every block
-- Prefer no indentation for `private`, `public`, and `protected` statements
-- Each variable declaration should have it's own type declaration
+private:
+  int memeber_var_; 
+}
+```
+## Spacing
```

Comment:
I think it's better to mention no trailing whitespace.

Reply:
as in
```c++
int member_var_  ;
```
?

Reply:
For example
int member_var_;xxx
where xxx are the extra whitespace.
We have many this kind of whitespace in the current code.

Reply:
I see, good suggestion, I've updated the doc